homeassistant/components/roborock/sensor.py

```diff
@@ -0,0 +1,129 @@
+"""Support for Roborock sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from roborock.typing import RoborockDeviceProp
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.util import slugify
+
+from .const import (
+    DOMAIN,
+    FILTER_REPLACE_TIME,
+    MAIN_BRUSH_REPLACE_TIME,
+    SENSOR_DIRTY_REPLACE_TIME,
+    SIDE_BRUSH_REPLACE_TIME,
+)
+from .coordinator import RoborockDataUpdateCoordinator
+from .device import RoborockCoordinatedEntity
+from .models import RoborockHassDeviceInfo
+
+
+@dataclass
+class RoborockSensorDescriptionMixin:
+    """A class that describes sensor entities."""
+
+    value_fn: Callable[[RoborockDeviceProp], int]
+
+
+@dataclass
+class RoborockSensorDescription(
+    SensorEntityDescription, RoborockSensorDescriptionMixin
+):
+    """A class that describes Roborock sensors."""
+
+
+CONSUMABLE_SENSORS = [
+    RoborockSensorDescription(
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        key="main_brush_work_time",
+        icon="mdi:brush",
+        device_class=SensorDeviceClass.DURATION,
+        translation_key="main_brush_left",
+        value_fn=lambda data: MAIN_BRUSH_REPLACE_TIME
+        - data.consumable.main_brush_work_time,
```

Comment:
We should not show the time left. That is a calculated value. Instead, we should use the values provided by the device.

Reply:
I assume the fact that the app shows the user the calculated values doesn't change anything?

Roborock recommends replacing the consumables every x hours - as far as data goes, the time left is more useful than the time used as most people won't remember the amount of time that a consumable can be used for before replacing.

![image](https://user-images.githubusercontent.com/20257911/234550122-89f792bb-06b3-4935-82ce-0ff86621fe6d.jpeg)

I'm fine with changing it to the exposed value, I just wanted to ask for clarification sake

Thanks!

Reply:
I will discuss this a with some other core devs later this week. Will get back to this.

Reply:
I went ahead and made the change - and if you think it is best to just do work_time instead of time left, then I am all good with that!

Reply:
@Lash-L The main point is that we don't want to have calculations in the home assistant integration. If you think time left is more suitable, you should update the library to provide that. If work_time is good, that works too.

Reply:
It is now being done in the library